confgenerator/logging_processors.go

```diff
@@ -126,6 +212,45 @@ func (p LoggingProcessorParseJson) Components(ctx context.Context, tag, uid stri
 	return parserFilters
 }
 
+func (p LoggingProcessorParseJson) Processors(ctx context.Context) ([]otel.Component, error) {
+	from := p.Field
+	if from == "" {
+		from = "jsonPayload.message"
+	}
+	m, err := filter.NewMemberLegacy(from)
+	if err != nil {
+		return nil, err
+	}
+
+	fromAccessor, err := m.OTTLAccessor()
+	if err != nil {
+		return nil, err
+	}
+
+	statements := ottl.NewStatements(
+		ottl.LValue{"body"}.SetIf(ottl.ParseJSON(fromAccessor), fromAccessor.IsPresent()),
```

Comment:
This produces a runtime error and drops the log if the incoming log has a string-valued `Body`:

```
Jan 18 11:39:00 ops-agent otelopscol[11652]: 2024-01-18T11:39:00.197-0500#011error#011logs/processor.go:54#011failed processing logs#011{"error": "failed to execute statement: set(body, ParseJSON(body[\"message\"])) where (body != nil and body[\"message\"] != n
il), log bodies of type Str cannot be indexed"}
```

The log shouldn't be dropped if it can't be parsed, but beyond that, there should also be a supported way to parse JSON out of a string-valued body. The idea that immediately comes to mind is that we should allow `parse_json` to use the `textPayload` field since that's where a string-valued body would end up in the LogEntry.

We should fix the dropped logs in this PR but we can defer supporting `textPayload` to a later PR.

Reply:
I think the existing code would actually work if you configured `field: jsonPayload` on the processor. We have to default to `jsonPayload.message` for backwards compatibility.

As far as runtime errors go, I already have a TODO in this PR to make sure that we ignore parse failures. And I can add `textPayload` as an alias for `body`, but I think that's about the best I can do. We can't make `parse_json` automagically work because we already promised it would look for `jsonPayload.message`.

Reply:
`field: jsonPayload` feels weird since users will see the log in `textPayload` if left unparsed. Agreed that we shouldn't change the default behaviour when `field` is left unspecified, I actually meant allowing users to specify `field: textPayload` (which I think is consistent with your idea of aliasing `textPayload` to `body`).

Reply:
I've added `textPayload` as a valid `field`.